trustmanager/x509utils.go

```diff
@@ -125,8 +95,19 @@ func fingerprintCert(cert *x509.Certificate) CertID {
 	block := pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw}
 	pemdata := pem.EncodeToMemory(&block)
 
+	keyType := ""
+	switch cert.PublicKeyAlgorithm {
+	case x509.RSA:
+		keyType = "RSA"
+	case x509.ECDSA:
+		keyType = "ECDSA"
+	default:
+		logrus.Debug("error while fingerprinting certificate. Got Unknown key type.")
```

Comment:
I had to rework a ton of things that expected this method to never error out, but done.